tests/unit/amo/components/TestCollectionManager.js

```diff
@@ -334,6 +336,45 @@ describe(__filename, () => {
     );
   });
 
+  it('creates a collection with an add-on on submit', () => {
+    const errorHandler = createStubErrorHandler();
+    const id = 123;
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const root = render({
+      collection: null,
+      creating: true,
+      errorHandler,
+      router: createFakeRouter({
+        location: fakeRouterLocation({ query: { include_addon_id: id } }),
+      }),
+    });
+
+    // Fill in the form with values.
+    const name = 'A collection name';
+    const description = 'A collection description';
+    const slug = 'collection-slug';
+
+    typeInput({ root, name: 'name', text: name });
+    typeInput({ root, name: 'description', text: description });
+    typeInput({ root, name: 'slug', text: slug });
+
+    simulateSubmit(root);
+
+    sinon.assert.calledWith(
+      dispatchSpy,
+      createCollection({
+        defaultLocale: lang,
+        description: { [lang]: description },
+        errorHandlerId: errorHandler.id,
```

Comment:
This could also be `errorHandlerId: root.instance().props.errorHandler.id` to save yourself some setup.